### DISCUSSION TITLE
automated emails post limit time based

### DISCUSSION THREAD

**Discussion by X-Raym** [hizzle-co/noptin#576] [Ideas]
Hi,

Currently, automated emails query last 10 posts. The count can be overriden by shortcode attribute `limit`.

It would be nice if limit could be time based. For eg, if post automation is everyday, then only posts of that day. if it is by months, then only posts of this months. if it is every two weeks etc...

Meanwhile I will try to filter `$campaign_post` in the automated email template.

Thx!

**Comment by picocodes** [comment]
That's what happens. If the automation is set to run every day and you only publish 2 posts then only 2 posts will appear even if you set the limit to 10.

You can set -1 as the limit to send all posts published in the preceding period. [Take a look at this section of our docs for more details](https://noptin.com/guide/sending-emails/new-post-notifications/#limit-post-count).

**Comment by X-Raym** [comment]
@picocodes I'm not sure this is what happen, I had to code certain things to add this.  Where is the code supposed to handle this?

**Comment by X-Raym** [comment]
@picocodes Oh my first newsletter came out and I see what went wrong...

It seems the preview button in email campaign list doesnt display what will be sent 🥴

Here is from preview link in automation page (post limits set to 25, filtered by code on template page):
https://cloud.extremraym.com/realinks-newsletter-preview-from-campaign.png

Here here is the preview link from the sent emails post list and how it has been set (post limit set to every week, posts are from this week):
https://cloud.extremraym.com/realinks-newsletter-sent.png

So my first newsletter has been set truncated :/

Should I opend an issue ?

**Comment by X-Raym** [comment]
The Send test emails match the preiew link from automation page, not what was already either

**Comment by picocodes** [comment]
The test and preview functions are only there to give you a general idea of how your email will look. That's why they display posts even if they were published 10 years ago.

As stated in my previous comment, the actual email only lists posts that we published in the preceding period. That means if you only published 5 posts in the previous 7 days, and you've set up a weekly newsletter then only 5 posts will show up. If you don't publish any posts then the email won't send.

This is how post-digests work.

**Comment by X-Raym** [comment]
@picocodes Yes I figured this out but it was too late, I set my newsletter based on what was on preview. I had to inspect code and find the right filters, the right functions and template to see how it would behave, and adjust limit based on that (my first newsletter needed more posts that the period)... and something was sent.

So if you consider preview not showing what will be sent as a bug, please consider it as a feature request, cause there is currently no way to know that it will not work as planned, and having it showing what will be sent exactly will definitely be a lot more helpful and avoid surprises.
It seems it is just about making the preview take the time period into account, so almost all is here to have preview match what will be sent. 😉

Many thanks for your consideration!
